daemon/cluster/executor/container/controller.go

```diff
@@ -448,14 +475,19 @@ func (r *controller) Logs(ctx context.Context, publisher exec.LogPublisher, opti
 		return errors.Wrap(err, "failed getting container logs")
 	}
 
+	// use r.Task which is concurrency safe
+	task, tErr := r.Task()
+	if tErr != nil {
+		return errors.Wrap(err, "error getting controller task")
```

Comment:
these are different cases. the `tErr` is if for some reason `r.Task()` returns an error (it doesn't, this just guards against possible future changes because the method signature allows it to)

the `err` is the error returned from trying to inspect the container.